iceoryx_examples/benchmark_optional_and_expected/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+# Build icedelivery example
+cmake_minimum_required(VERSION 3.5)
+project(benchmark_optional_and_expected)
+
+include(GNUInstallDirs)
+
+find_package(iceoryx_utils CONFIG REQUIRED)
+
+get_target_property(ICEORYX_CXX_STANDARD iceoryx_utils::iceoryx_utils CXX_STANDARD)
+if ( NOT ICEORYX_CXX_STANDARD )
+    include(IceoryxPlatformDetection)
+endif ( NOT ICEORYX_CXX_STANDARD )
+
+add_executable(benchmark_optional_and_expected ./benchmark_optional_and_expected.cpp)
```

Comment:
Can you use `-` as the separator in the target name.
I am merging in the new `iox-convention` in the next PR (as discussed).
Also, prefix with `iox`.

Reply:
we have to be careful with this, there are internal scripsts which might to be adjusted ... and more importantly we need to communicate this change to our users